components/cli/ballerina/docker_bal_executor.go

```diff
@@ -0,0 +1,147 @@
+	if err != nil {
+		return fmt.Errorf("error in determining working directory, %v", err)
+	}
+	// Retrieve the cellery cli docker instance status.
+	cmdDockerPs := exec.Command("docker", "ps", "--filter",
+		"label=ballerina-runtime="+version.BuildVersion(),
+		"--filter", "label=currentDir="+currentDir, "--filter", "status=running", "--format", "{{.ID}}")
+	containerId, err := cmdDockerPs.Output()
+	if err != nil {
+		return fmt.Errorf("error in retrieving cellery cli docker instance status, %v", err)
+	}
+
+	if string(containerId) == "" {
+		cmdDockerRun := exec.Command("docker", "run", "-d",
+			"-l", "ballerina-runtime="+version.BuildVersion(),
+			"-l", "current.dir="+currentDir,
+			"--mount", "type=bind,source="+currentDir+",target=/home/cellery/src",
+			"--mount", "type=bind,source="+util.UserHomeDir()+string(os.PathSeparator)+".ballerina,target=/home/cellery/.ballerina",
+			"--mount", "type=bind,source="+util.UserHomeDir()+string(os.PathSeparator)+".cellery,target=/home/cellery/.cellery",
+			"wso2cellery/ballerina-runtime:"+version.BuildVersion(), "sleep", "600",
+		)
+		stderrReader, err := cmdDockerRun.StderrPipe()
+		if err != nil {
+			return fmt.Errorf("error while building stderr pipe, %v", err)
+		}
+		stdoutReader, _ := cmdDockerRun.StdoutPipe()
+		if err != nil {
+			return fmt.Errorf("error while building stdout pipe, %v", err)
+		}
+		stderrScanner := bufio.NewScanner(stderrReader)
+		stdoutScanner := bufio.NewScanner(stdoutReader)
+		err = cmdDockerRun.Start()
+		if err != nil {
+			return fmt.Errorf("error while starting docker process, %v", err)
+		}
+		go func() {
+			for {
+				if stderrScanner.Scan() && strings.HasPrefix(stderrScanner.Text(), "Unable to find image") {
+					break
+				}
+			}
+		}()
+		go func() {
+			for {
+				if stdoutScanner.Scan() {
+					containerId = []byte(stdoutScanner.Text())
+					break
+				}
+			}
+		}()
+
+		err = cmdDockerRun.Wait()
+		if err != nil {
+			return fmt.Errorf("error while running ballerina-runtime docker image, %v", err)
+		}
+		time.Sleep(5 * time.Second)
+	}
+
+	cliUser, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("error while retrieving the current user, %v", err)
+	}
+	if cliUser.Uid != constants.CELLERY_DOCKER_CLI_USER_ID {
+		cmdUserExist := exec.Command("docker", "exec", strings.TrimSpace(string(containerId)),
+			"id", "-u", cliUser.Username)
+		_, errUserExist := cmdUserExist.Output()
+		if errUserExist != nil {
+			cmdUserAdd := exec.Command("docker", "exec", strings.TrimSpace(string(containerId)), "useradd", "-m",
+				"-d", "/home/cellery", "--uid", cliUser.Uid, cliUser.Username)
+
+			_, errUserAdd := cmdUserAdd.Output()
+			if errUserAdd != nil {
+				return fmt.Errorf("error in adding Cellery execution user, %v", err)
+			}
+		}
+	}
+	re := regexp.MustCompile("^" + currentDir + "/")
+	balFilePath := re.ReplaceAllString(fileName, "")
+	cmd := exec.Command("docker", "exec", "-w", "/home/cellery/src", "-u", cliUser.Uid,
```

Comment:
move /home/cellery to a const and refer from all places hardcoded.

Reply:
Updated.